server/src/main/java/io/crate/planner/operators/LogicalPlanner.java

```diff
@@ -450,44 +489,46 @@ public LogicalPlan visitUnionSelect(UnionSelect union, List<Symbol> outputs) {
             );
         }
 
+        private static List<Symbol> mergeOutputs(AnalyzedRelation relation, List<Symbol> outputs) {
+            LinkedHashSet<Symbol> result = new LinkedHashSet<>(relation.outputs());
+            SequencedSet<RelationName> relationNamesFromRelation = RelationNames.getDeep(relation);
+            Predicate<Symbol> addFiltered = node -> {
+                if (node instanceof ScopedSymbol || node instanceof Reference) {
+                    SequencedSet<RelationName> relationNamesFromSymbol = RelationNames.getDeep(node);
+                    for (RelationName relationName : relationNamesFromSymbol) {
+                        if (relationNamesFromRelation.contains(relationName)) {
+                            result.add(node);
+                            break;
+                        }
+                    }
+                }
+                return false;
+            };
+            for (Symbol symbol : outputs) {
+                symbol.any(addFiltered);
+            }
+            return List.copyOf(result);
+        }
+
         @Override
         public LogicalPlan visitQueriedSelectRelation(QueriedSelectRelation relation, List<Symbol> outputs) {
             SplitPoints splitPoints = SplitPointsBuilder.create(relation);
             SubQueries subQueries = subqueryPlanner.planSubQueries(relation);
-            LogicalPlan source = JoinPlanBuilder.buildJoinTree(
+            LogicalPlan source = buildImplicitJoins(
                 relation.from(),
                 relation.where(),
-                relation.joinPairs(),
                 subQueries,
                 rel -> {
+                    List<Symbol> toCollect = splitPoints.toCollect();
                     if (relation.from().size() == 1) {
-                        return rel.accept(this, splitPoints.toCollect());
+                        return rel.accept(this, toCollect);
                     } else {
                         // Need to pass along the `splitPoints.toCollect` symbols to the relation the symbols belong to
                         // We could get rid of `SplitPoints` and the logic here if we
                         // a) introduce a column pruning
                         // b) Make sure tableRelations contain all columns (incl. sys-columns) in `outputs`
```

Comment:
Maybe move this comment to `mergeOutputs`  ? Seems relevant for the other callsites as well.